docs/configuration.asciidoc

```diff
@@ -834,6 +834,16 @@ Example:
 ELASTIC_APM_GLOBAL_LABELS="subspace=sap-hana,rack=number6"
 ----
 
+[[extended-label-types]]
+==== `extendedLabelTypes`
+
+* *Type:* Boolean
+* *Default:* `false`
+* *Env:* `ELASTIC_APM_EXTENDED_LABEL_TYPES`
+
+Whether numeric and boolean labels should be sent as-is to the APM server.
+When this option is `false`, all labels are sent as strings, no matter their original type.
+
```

Comment:
Did you consider adding an optional argument to `setLabel` and `addLabels`, something like:

```diff
-GenericSpan.prototype.setLabel = function (key, value) {
+// TODO: in v4, remove the stringify option and never stringify
+GenericSpan.prototype.setLabel = function (key, value, stringify = true) {
```

Pros that I see:
- The user of `setLabel()` needs to explicitly work through each usage in their code to verify that changing 'stringify' won't break them.
- One isn't led to infer that `extendedLabelTypes=false` might affect the `globalLabels` config var (it won't, globalLabels already passes through bools and numbers untouched).

Cons:
- The user may need to slog through multiple usages. :)

Reply:
I didn't TBH. I very much like the way you can configure the agent with env vars, like most containing apps do anyway... I thought users could
1. Update the app environment, restart the app
2. Let the agent send a few transactions
3. Revert the environment, restart the app

Then they can safely test their label-specific ES queries on the new transactions and validate that the type changes work.

But I'm OK to move to an explicit third parameter and drop the config flag, you tell me 👍

Reply:
> But I'm OK to move to an explicit third parameter and drop the config flag

Yes please, and thanks. My initial gut reaction is to prefer the extra param to `setLabel` and `addLabels`. I asked a colleague and he preferred the extra param as well (more strongly than I).